app/raid.js

```diff
@@ -1,225 +1,294 @@
-			if (this.roles.mystic && member.roles.has(this.roles.mystic.id)) {
-				attendees_list += ' <:mystic:346183029171159041>';
-			} else if (this.roles.valor && member.roles.has(this.roles.valor.id)) {
-				attendees_list += ' <:valor:346182738652561408>';
-			} else if (this.roles.instinct && member.roles.has(this.roles.instinct.id)) {
-				attendees_list += ' <:instinct:346182737566105600>';
-			}
-
-			attendees_list += '\n';
-		}
-
-		return {
-			"embed": {
-				"title": `Level 5 Raid against ${pokemon}`,
-				"description": `Raid available until ${end_time}\n` +
-								`${location}\n` +
-								`Join this raid by typing the command \`\`\`!join ${raid_data.id}\`\`\`\n\n` +
-								`Potential Trainers _(${total_attendees} total)_:\n` +
-								`${attendees_list}\n` +
-								`Starting @ **${((raid_data.start_time)? (raid_data.start_time): '????')}**\n`,
-				"url": (location)? location: 'https://discordapp.com',
-				"color": 4437377,
-				"thumbnail": {
-					"url": "https://rankedboost.com/wp-content/plugins/ice/pokemon-go/" + pokemon + "-Pokemon-Go.png"
-				},
-				// "author": {
-				// 	"name": "author name",
-				// 	"url": "https://discordapp.com",
-				// 	"icon_url": "https://cdn.discordapp.com/embed/avatars/0.png"
-				// },
-				// "fields": [
-				// 	{
-				// 		"name": raid_data.attendees.length + " will be attending @ " + ((raid_data.start_time)? (raid_data.start_time): '????'),
-				// 		"value": attendees_list
-				// 	}
-				// ],
-				// "footer": {
-				// 	"text": (raid_data.start_time)? "Raid Begining @ " + raid_data.start_time: "Still determining a start time..."
-				// }
-			}
-		};
-	}
+    // message.member.displayName, message.guild
+    raid_data.attendees.push(member);
+
+    this.setUserRaidId(member, raid_id);
+
+    return {raid: raid_data};
+  }
+
+  removeAttendee(channel, member, raid_id) {
+    const raid_data = this.getRaid(channel, member, raid_id);
+
+    // message.member.displayName, message.guild
+    const index = raid_data.attendees.findIndex((m) => {
+      return m.id === member.id;
+    });
+
+    raid_data.attendees.splice(index, 1);
+
+    this.setUserRaidId(member, raid_id);
+
+    return {raid: raid_data};
+  }
+
+  setArrivalStatus(channel, member, raid_id, status) {
+    const raid_data = this.getRaid(channel, member, raid_id);
+
+    for (let i = 0; i < raid_data.attendees.length; i++) {
+      let m = raid_data.attendees[i];
+
+      // TODO:  Can't set arrived status on member as it is on the MEMBER and thus will be set on other raids they attend
+      //			need to save some where else, and need to save the main "author" as the raid leader for masterball status
+      if (m.id === member.id) {
+        m.has_arrived = status;
+        break;
+      }
+    }
+
+    this.setUserRaidId(member, raid_id);
+
+    return {raid: raid_data};
+  }
+
+  setRaidTime(channel, member, raid_id, start_time) {
+    const raid_data = this.getRaid(channel, member, raid_id);
```

Comment:
ok, so I don't pretend to know everything about javascript, but shouldn't this throw an error? I thought const variables can't change, but 2 lines down I'm changing the start time.... this seems like it should be a problem?